sql/rpl_gtid.cc

```diff
@@ -849,9 +861,20 @@ rpl_slave_state::gtid_delete_pending(THD *thd,
 
 #ifdef WITH_WSREP
   /*
-    Updates in slave state table should not be appended to galera transaction
-    writeset.
+    We should replicate local gtid_slave_pos updates to other nodes.
+    In applier we should not append them to galera writeset.
   */
+  if (WSREP_ON_ && wsrep_thd_is_local(thd) &&
+      thd->wsrep_cs().state() != wsrep::client_state::s_none)
```

Comment:
is wsrep_thd_is_local() used here to filter for async replication SQL thread execution? IF so, it would be better to filter for THD's thread type of: SYSTEM_THREAD_SLAVE_SQL

Reply:
No, consider case master ---async replication ---> galera slave ----galera replication ---> galera node. Idea is that galera slave should replicate these deletes to another galera node. But in applier we should not anymore replicate them i.e. only local deletes are replicated.

Reply:
wsrep_thd_is_local(thd) filters out wsrep applier threads, and lets through local connections. Async replication slave SQL thread is treated as local connection as well, so this if block is executed for local connections and replication slave SQL thread